jenkins_pipelines/scripts/SUSEManager_cleaner/suse_manager_cleaner_program/suse_manager_api.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import xmlrpc.client
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Global configuration
+username = "admin"
+password = "admin"
+
+class ResourceManager:
+    def __init__(self, manager_url, resources_to_delete, product_version):
+        self.manager_url = manager_url
+        self.resources_to_delete = {"proxy", "monitoring", "build"} - set(resources_to_delete)
+        self.product_version = product_version
+        self.client = None
+        self.session_key = None
+
+    def get_session_key(self):
+        self.client = xmlrpc.client.ServerProxy(f"http://{self.manager_url}/rpc/api")
+        self.session_key = self.client.auth.login(username, password)
+        logger.info("Session key obtained.")
+
+    def logout_session(self):
+        self.client.auth.logout(self.session_key)
+        logger.info("Logged out from session.")
+
+    def delete_users(self):
+        users = self.client.user.listUsers(self.session_key)
+        for user in users:
+            if user["login"] != "admin":
+                logger.info(f"Delete user: {user['login']}")
+                self.client.user.delete(self.session_key, user["login"])
+
+    def delete_activation_keys(self):
+        activation_keys = self.client.activationkey.listActivationKeys(self.session_key)
+        for activation_key in activation_keys:
+            if not any(protected in activation_key['key'] for protected in self.resources_to_delete):
+                logger.info(f"Delete activation key: {activation_key['key']}")
+                self.client.activationkey.delete(self.session_key, activation_key['key'])
+
+    def delete_config_projects(self):
+        projects = self.client.contentmanagement.listProjects(self.session_key)
+        for project in projects:
+            logger.info(f"Delete project: {project['label']}")
+            self.client.contentmanagement.removeProject(self.session_key, project['label'])
+
+    def delete_software_channels(self):
+        channels = self.client.channel.listMyChannels(self.session_key)
+
+        if self.product_version == "uyuni":
+            channels = self.client.channel.listMyChannels(self.session_key)
+            for channel in channels:
+                if "custom" in channel['label'] and not any(protected in channel['label'] for protected in self.resources_to_delete):
```

Comment:
Not strictly necessary and can always be done later on, as refactor/improvement.

We can probably have a private function to delete channels that takes a few parameters and replaces this code blocks.